### PR TITLE
Code changes to support Jkarni/content types.

### PR DESCRIPTION
**Pull request by tvh**

### CODE REVIEW

**Comment by thsutton** [comment]
@tvh It would be nice to add some additional types to the API in the test.

File: src/Servant/JQuery/Internal.hs [hunk 1]
```diff
@@ -157,7 +161,12 @@ paramToStr qarg notTheEnd =
 defReq :: AjaxReq
 defReq = AjaxReq defUrl "GET" [] False ""
 
-class HasJQ layout where
+type family Elem (a :: *) (ls::[*]) :: Constraint where
+  Elem a '[] = 'False ~ 'True
```
Comment (christian-marie) [inline src/Servant/JQuery/Internal.hs]: Does this work without this? 'False ~ 'True is such a bad type error if it does what I think it does.

Comment (jkarni) [inline src/Servant/JQuery/Internal.hs]: I just realized we could be using this for better type errors. E.g.
```
type TheExpectedPreConditionDescribed = False
...
Elem a '[] = 'TheExpectedPreConditionDescribed ~ 'True
```
or some such thing.

**Comment by jkarni** [comment]
LGTM. Let's wait for https://github.com/haskell-servant/servant/pull/9#discussion_r24979301 before merging though.

**Comment by alpmestan** [comment]
Nice, thanks. I'm just a bit sad we have to introduce `UndecidableInstances` there.

**Comment by jkarni** [comment]
@tvh if you rebase against master, the build should pass, and we'll merge.

**Pull request merged**